.github/workflows/reusable-update-pre-commit.yml

```diff
@@ -0,0 +1,49 @@
+name: Update pre-commit
+# Update pre-commit config and create PR if changes are detected
+# author: Christoph Fröhlich <christoph.froehlich@ait.ac.at>
+
+on:
+  workflow_call:
+
+jobs:
+  auto_update_and_create_pr:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install pre-commit
+        run: |
+          pip install pre-commit
+
+      - name: Auto-update with pre-commit
+        run: |
+          pre-commit autoupdate || true  # Ignoring errors
+
+      - name: Check for changes
+        id: git_status
+        run: |
+          git diff --quiet && echo "::set-output name=changed::false" || echo "::set-output name=changed::true"
+
+      - name: There are changes
+        if: steps.git_status.outputs.changed == 'true'
+        run: |
+          echo "Files have changed"
+          git diff --exit-code || true
+
+      - name: No changes!
+        if: steps.git_status.outputs.changed == 'false'
+        run: |
+          echo "No changes detected"
+
+      - name: Create Pull Request
+        if: steps.git_status.outputs.changed == 'true'
+        uses: peter-evans/create-pull-request@v6
```

Comment:
Would we want to specify the author here, as well? Otherwise, the default author [Defaults to the user who triggered the workflow run](https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#action-inputs) which might be misleading in some situations.

Reply:
ah, you are right: https://github.com/orgs/community/discussions/47855#discussioncomment-7061183

Reply:
I'd test this first without specifying it, because in the demos of the action the PR is also from the github action bot without having specified it.. let's see ;)